pkg/apis/certmanager/v1/types_certificate.go

```diff
@@ -205,10 +205,7 @@ type CertificatePrivateKey struct {
 	// If `algorithm` is set to `ECDSA`, valid values are `256`, `384` or `521`,
 	// and will default to `256` if not specified.
 	// No other values are allowed.
-	// +kubebuilder:validation:ExclusiveMaximum=false
-	// +kubebuilder:validation:Maximum=8192
-	// +kubebuilder:validation:ExclusiveMinimum=false
-	// +kubebuilder:validation:Minimum=0
+	// Size validation is done by the validating webhook.
```

Comment:
It's a nitpick, but I don't think this sentence belongs in the API docs. 
The user who runs `kubectl explain certificate` shouldn't need to know about the validation mechanism.
They only need to know what are valid values, which is already explained above.

But the comment is useful for developers who might be tempted to re-instate the openapi validation here.
I wonder if there's a way to add a comment which doesn't get added to the CRD documentation?

And I wonder if the comment should be more explicit; 
> DO NOT ADD OPENAPI VALIDATION BECAUSE IT BREAKS HELM. See ....

Or we could add a note to that effect to the contributing docs on the website.
And perhaps we need to prioritise those Helm upgrade tests, so that we catch these problems earlier.
Please create an issue if there isn't already one.

Reply:
> I wonder if there's a way to add a comment which doesn't get added to the CRD documentation?
And I wonder if the comment should be more explicit

Agreed. I did not find any `openapi-gen`- specific way to do it, but adding a comment to the same line works (although makes it very long).
`Size int `json:"size,omitempty"` // Validated by webhook. Be mindful of adding OpenAPI validation- see https://github.com/jetstack/cert-manager/issues/3644 .`
I feel like it's a better fit for being in the code rather than contributing docs as it's quite specific to this bit of code only, but might be wrong.

Reply:
> And perhaps we need to prioritise those Helm upgrade tests, so that we catch these problems earlier

I've created #3650 for testing Helm installation/upgrade. I wonder if it would not make sense to consider separating the process of building/testing the binaries from the process of creating/testing the Helm charts at some point?